ui-tests/src/test/java/io/jenkins/plugins/coverage/CoveragePublisherTest.java

```diff
@@ -0,0 +1,135 @@
+package io.jenkins.plugins.coverage;
+
+import org.junit.Ignore;
+import org.junit.Test;
+
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+
+import io.jenkins.plugins.coverage.CoveragePublisher.CoveragePublisher.SourceFileResolver;
+import io.jenkins.plugins.coverage.util.ChartUtil;
+
+import static net.javacrumbs.jsonunit.assertj.JsonAssertions.*;
+import static org.assertj.core.api.AssertionsForClassTypes.*;
+
+/**
+ * Acceptance tests for CoveragePublisher. Verifies if set options in CoveragePublisher are used and lead to excepted
+ * results.
+ */
+public class CoveragePublisherTest extends UiTest {
+    /**
+     * Verifies that job with no report fails when setFailNoReports(true).
+     */
+    @Test
+    public void testFailOnNoReport() {
+        FreeStyleJob job = getJobWithoutAnyReports(InCaseNoReportsConfiguration.FAIL);
+        buildWithErrors(job);
+    }
+
+    /**
+     * Verifies that job with decreased coverage fails when setFailBuildIfCoverageDecreasedInChangeRequest(true).
+     */
+    @Test
+    public void testFailOnDecreasedCoverage() {
+        FreeStyleJob job = getJobWithFirstBuildAndDifferentReports(InCaseCoverageDecreasedConfiguration.FAIL);
+        buildWithErrors(job);
+    }
+
+    /**
+     * Test if build fails if setFailUnhealthy is true and thresholds set.
+     */
+    @Test @Ignore("This bug needs to be fixed")
+    public void testAdapterThresholdsAndFailOnUnhealthySetter() {
+        FreeStyleJob job = getJobWithAdapterThresholdAndFailOnUnhealthySetter(97, 99, true, ThresholdLevel.ADAPTER);
+        buildWithErrors(job);
+    }
+
+    /**
+     * Test if global thresholds are set.
+     */
+    @Test
+    public void testGlobalThresholdsAndFailSetter() {
+        FreeStyleJob job = getJobWithAdapterThresholdAndFailOnUnhealthySetter(97, 99, true, ThresholdLevel.GLOBAL);
+        buildUnstable(job);
+    }
+
+    /**
+     * Tests if source file storing level and display is correct.
+     */
+    @Test @WithPlugins("git")
+    public void testSourceFileStoringLevelAllBuilds() {
+        FreeStyleJob job = getJobWithReportAndSourceCode(SourceFileResolver.STORE_ALL_BUILD);
```

Comment:
It might make more sense to bundle one source code file with the test data so Git is not necessary. (See https://github.com/jenkinsci/code-coverage-api-plugin/blob/master/plugin/src/test/java/io/jenkins/plugins/coverage/model/CoveragePluginSourceITest.java)